### PR TITLE
refactor to functional component & use hooks

### PR DESCRIPTION
**Pull request by rebaz36**
# In This PR i implemented the followings

- refactor class based components to functional components
- refactor components to use hooks

### CODE REVIEW

File: src/component/Calculator.js [hunk 1]
```diff
@@ -1,63 +1,68 @@
...
+export default function Calculator() {
+  const [state, setState] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
+
+  const handleClick = (e) => {
+    // const result = calculate(this.state, e.target.textContent);
+    // this.setState(result);
```
Comment (danmainah) [inline src/component/Calculator.js]: - [ ] Kindy remove the commented code, it's not good practice to have commented code in your codebase.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @rebaz36 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [x] Kindly change the pull request description from we to I since this is an individual project.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/component/Calculator.js (hunk 1 above)
Comment (rebaz36) [reply src/component/Calculator.js]: thanks for the tip @danmainah

**Review by BertrandConxy — APPROVED** [review]
Hi @rebaz36, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💪 
- No linter errors
- project requirements are met
- project works fine
- good PR description
- correct flow is used

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rebaz36** [comment]
thanks @BertrandConxy

**Pull request merged**